backend/internal/sessions/sessions.go

```diff
@@ -76,6 +76,11 @@ func (s *sessions) List(ctx context.Context, filter sdkservices.ListSessionsFilt
 	return s.svcs.DB.ListSessions(ctx, filter)
 }
 
+func (s *sessions) Delete(ctx context.Context, sessionID sdktypes.SessionID) error {
+	s.z.With(zap.String("session_id", sessionID.String())).Debug("delete")
```

Comment:
I think it is better to log what happened and not what about to happen
otherwise you can see delete in log, but service can fail and you won't know if it happened or not
they deleting can also fail, so the log might indicate something other than intended

but that's my taste for logging (unless you verbose log everything searching for a bug)

Reply:
depends.. This log before will cover both cases - success and failure. On success you will see that session deletion was triggered and on failure you will see the trigger and error logging on deeper level functions.
Better and cleaner solution may be call this function, cache err, check and log. But it maybe overkill for our current purposes, e.g. 
```go
   if err := s.svcs.DB.DeleteSession(ctx, sessionID); err != nil {
      s.z.With(zap.String("session_id", sessionID.String())).Debug("delete")
   } else {
      s.z.With(zap.String("session_id", sessionID.String()), zap.Error(err)).Debug("delete failed")
   }
```
maybe we consider to create some kind of generic wrapper func for this.
WDYT?

Reply:
in the extreme case that the process will fail after logging delete and before actually deleting it (calling the database), you will get invalid logs

in the not so uncommon case, where by a bug or mistake, the error isn't printed to log after the delete fail, you still have an invalid log

so for me, as a guideline, I try to log what happened and not what I'm about to do, at least on info level

debug logs can be as verbose as I need, but most of the times, the log level in production is info and not debug

so maybe, info should indicate what happened, and debug can do what ever required ?

Reply:
Agree. Above log is DEBUG btw.
resolving?